shared/actions/tracker.js

```diff
@@ -296,10 +316,34 @@ export function onFollow (username: string, localIgnore: bool): (dispatch: Dispa
   }
 }
 
-export function onClose (username: string): Action {
-  return {
-    type: Constants.onClose,
-    payload: {username}
+function _dismissWithToken (trackToken) {
+  const params : track_dismissWithToken_rpc = {
+    method: 'track.dismissWithToken',
+    param: {trackToken},
+    incomingCallMap: {},
+    callback: err => {
+      if (err) {
+        console.log('err dismissWithToken', err)
+      }
+    }
+  }
+  engine.rpc(params)
+}
+
+export function onClose (username: string): TrackerActionCreator {
+  return (dispatch, getState) => {
+    const trackToken = _getTrackToken(getState, username)
+
+    if (trackToken) {
```

Comment:
i dislike that its unbalanced where we set it in the reducer but unset it in the action. this is a problem with redux where the reducer itself can't spawn actions. i figured this is better than having onClose also dispatch just setting this variable. i think ultimately sagas are meant to fix this kind of thing and we just don't have that in yet